### ISSUE TITLE
bundle node and pyright with the pypi package

### ISSUE THREAD

**Issue by DetachHead** [RobertCraigie/pyright-python#231]
having the pypi package download both node and the npm package when you first run the `pyright` command causes some issues:

- we have a docker image where all the dependencies are installed, but when the pyright job runs in our CI, it can fail because the runner cannot access npm (due to corporate proxy nonsense). to work around this, we had to put this command in our dockerfile:
  ```dockerfile
  # installs node & pyright
  RUN pyright --version
  ```
- #225
- npm is incredibly slow and unreliable on PCs at my work

i think bundling the npm package and node in the pypi package would be a much more reliable solution. that's how [the playwright pypi package works](https://pypi.org/project/playwright/) for example

**Comment by DetachHead** [comment]
I've solved this problem in [basedpyright](https://github.com/detachhead/basedpyright)

**Comment by Avasam** [comment]
Here's another example of issues caused by trying to install a node environment on first run: https://github.com/python/typeshed/pull/11575#pullrequestreview-1938595099

According to @jakebailey, https://github.com/python/typeshed/pull/11575#issuecomment-2000065260
> [...] It also could skip npm, but I can understand wanting to leverage npm to do version resolution (though doing it just with the registry is not super hard and IMO would be worth it to avoid the dep).

Also relates to #209

**Comment by RobertCraigie** [comment]
Didn't intent to close this yet, the next release will add support for
```
pip install pyright[nodejs]
```
which will download the Node.js binaries at install time using [`nodejs-wheel`](https://pypi.org/project/nodejs-wheel-binaries/) instead of using `nodeenv` at runtime.

**Comment by bersbersbers** [comment]
https://github.com/RobertCraigie/pyright-python/pull/299 works great for me - thanks!

**Comment by Puncher1** [comment]
For me, this doesn't fix it and sitll has the same behavior as described in #296, even when I installed it via
```
pip install pyright[nodejs]
```

**Comment by vctorhuang** [comment]
Hi @RobertCraigie, thanks for adding these great features and maintaining this repo! 

Our usage of `pyright` would optimally require that everything is bundled inside the pypi package (`node` + `pyright`). I built the wheel off of the branch (#300) according to the instructions above and it seems to work correctly. Is there a timeline on when this will be available on main?

**Comment by charlesnicholson** [comment]
Agreed, this would be very convenient for our workflows as well. Any ETA on landing this in a public release? Thanks for building and maintaining this, @RobertCraigie, it's extremely helpful!

**Comment by RobertCraigie** [comment]
Sorry for the delay here, this was released in `v1.1.392.post0`!

**Comment by winstxnhdw** [comment]
@RobertCraigie will you be removing the installation tip on installing the `nodejs` extra?

**Comment by RobertCraigie** [comment]
Not at the moment, the default behaviour is still downloading Node.js at runtime. The change I just made was bundling the `pyright` npm package.
